### PR TITLE
Add default tag for mixer

### PR DESCRIPTION
**Pull request by mandarjog**
1. Add mixerTag
2. Only delete namespaces that we create
3. Add `--skipBuild` option to e2e.sh

### CODE REVIEW

File: test/integration/driver.go [hunk 1]
```diff
@@ -146,6 +152,24 @@ func setup() {
 
 }
 
+func assertExistsNamespace(cl *kubernetes.Clientset, ns string) {
+	if nl, err := cl.Core().Namespaces().List(v1.ListOptions{}); err != nil {
```
Comment (kyessenov) [inline test/integration/driver.go]: Use Get(name) call directly.

File: test/integration/driver.go [hunk 2]
```diff
@@ -88,6 +90,8 @@ func init() {
 		"Docker hub")
 	flag.StringVarP(&tag, "tag", "t", "",
 		"Docker tag")
+	flag.StringVarP(&mixerTag, "mixerTag", "", "latest",
```
Comment (kyessenov) [inline test/integration/driver.go]: can we put a fixed (sha-based) tag? the test is supposed to be hermetic, and updates to mixer can accidentally break our tests.

File: test/integration/driver.go [hunk 3]
```diff
@@ -772,10 +798,15 @@ func generateNamespace(cl *kubernetes.Clientset) (string, error) {
 		return "", err
 	}
 	log.Printf("Created namespace %s\n", ns.Name)
+	cleanupNameSpace = true
 	return ns.Name, nil
 }
 
 func deleteNamespace(cl *kubernetes.Clientset, ns string) {
+	// do not cleanup namespace that we did not create
+	if !cleanupNameSpace {
```
Comment (kyessenov) [inline test/integration/driver.go]: move this to teardown() call to deleteNamespace

File: bin/e2e.sh [hunk 4]
```diff
@@ -21,16 +25,18 @@ done
 [[ ! -z "$hub" ]]       && args=$args" -h $hub"
 [[ ! -z "$namespace" ]] && args=$args" -n $namespace"
 
+args=$args" --mixerTag $mixerTag"
```
Comment (kyessenov) [inline bin/e2e.sh]: can you process mixerTag flag like in line 11-15? bin/e2e takes same arguments as driver and passes them along.

File: bin/e2e.sh [hunk 5]
```diff
@@ -21,16 +25,18 @@ done
...
-    bazel run //docker:$image$debug_suffix
-    docker tag istio/docker:$image$debug_suffix $hub/$image:$tag
-    docker push $hub/$image:$tag
+	bazel run //docker:$image$debug_suffix
+	docker tag istio/docker:$image$debug_suffix $hub/$image:$tag
+	docker push $hub/$image:$tag
 done
 
-bazel run //test/integration -- $args --norouting
+bazel $BAZEL_ARGS run //test/integration -- $args --norouting
```
Comment (kyessenov) [inline bin/e2e.sh]: what does this do?

File: test/integration/driver.go [hunk 6]
```diff
@@ -90,7 +91,7 @@ func init() {
 		"Docker hub")
 	flag.StringVarP(&tag, "tag", "t", "",
 		"Docker tag")
-	flag.StringVarP(&mixerTag, "mixerTag", "", "latest",
+	flag.StringVarP(&mixerTag, "mixerTag", "", "",
```
Comment (kyessenov) [inline test/integration/driver.go]: can you put SHA here too? I know it's duplication but I use both scripts e2e and driver.go for testing. driver.go bypasses docker image creation. We didn't find a good way to get rid of the shell script.

**Comment by mandarjog** [comment]
Looks like this is blocked on
https://github.com/istio/istio-testing/pull/71

**Pull request merged**